### ISSUE TITLE
Error: eth_abi.exceptions.InsufficientDataBytes: Tried to read 32 bytes, only got 0 bytes.

### ISSUE THREAD

**Issue by scottnoidea** [dipdup-io/dipdup#1022]
In the evm event indexer, the matcher fails on the below error when attempting to decode an event where the number of topics in the checked event is less than the number of topics in the ABI.

**Steps to reproduce:**

config export output:

```
advanced:
  alt_operation_matcher: false
  early_realtime: false
  postpone_jobs: false
  reindex: {}
  skip_version_check: false
  unsafe_sqlite: false
contracts:
  atoken_v2:
    abi: '0x1c050bca8babe53ef769d0d2e411f556e1a27e7b'
    kind: evm
    typename: atoken_v2
custom: {}
database:
  immune_tables: []
  kind: sqlite
  path: ':memory:'
datasources:
  etherscan:
    api_key: <key>
    kind: abi.etherscan
    url: https://api.etherscan.io/api
  evm_node:
    kind: evm.node
    rollback_depth: 32
    url: https://eth-mainnet.g.alchemy.com/v2/<key>
    ws_url: wss://eth-mainnet.g.alchemy.com/v2/<key>
  subsquid:
    kind: evm.subsquid
    node: evm_node
    url: https://v2.archive.subsquid.io/network/ethereum-mainnet
hooks: {}
indexes:
  ethereum_atoken_v2_events:
    datasource: subsquid
    first_level: 19915850
    last_level: 19915851
    handlers:
    - callback: on_mint
      contract: atoken_v2
      name: Mint
    kind: evm.subsquid.events
    last_level: 0
    node_only: false
    template_values: {}
jobs: {}
logging: INFO
package: ethereum
spec_version: '2.0'
templates: {}
```

**What did you expect to happen:**
Indexer should not match events which don't have the same number of topics as the ABI

**What actually happened:**

EVM event indexer throws error:

```
Traceback (most recent call last):
  File "/home/vscode/.local/bin/dipdup", line 8, in <module>
    sys.exit(cli())
             ^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 1205, in __call__
    return anyio.run(self._main, main, args, kwargs, **opts)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/anyio/_core/_eventloop.py", line 73, in run
    return async_backend.run(func, args, {}, backend_options)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 2001, in run
    return runner.run(wrapper())
           ^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/asyncio/base_events.py", line 653, in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/anyio/_backends/_asyncio.py", line 1989, in wrapper
    return await func(*args)
           ^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 1208, in _main
    return await main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 1120, in main
    rv = await self.invoke(ctx)
         ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 1739, in invoke
    return await _process_result(await sub_ctx.command.invoke(sub_ctx))
                                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 1485, in invoke
    return await ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/asyncclick/core.py", line 824, in invoke
    rv = await rv
         ^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/cli.py", line 127, in wrapper
    raise e
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/cli.py", line 120, in wrapper
    await fn(ctx, *args, **kwargs)
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/cli.py", line 289, in run
    await dipdup.run()
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/dipdup.py", line 686, in run
    await gather(*tasks)
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/dipdup.py", line 147, in run
    indexes_processed = any(await gather(*tasks))
                            ^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/index.py", line 224, in process
    await self._synchronize(sync_level)
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid.py", line 157, in _synchronize
    await self._synchronize_subsquid(sync_level)
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/index.py", line 56, in _synchronize_subsquid
    await self._process_level_data(events, sync_level)
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/index.py", line 121, in _process_level_data
    matched_handlers = self._match_level_data(self._config.handlers, level_data)  # type: ignore[union-attr]
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/index.py", line 105, in _match_level_data
    return match_events(self._ctx.package, handlers, level_data, self.topics)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/matcher.py", line 118, in match_events
    arg = prepare_event_handler_args(package, handler_config, event)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/matcher.py", line 74, in prepare_event_handler_args
    data = decode_event_data(
           ^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/matcher.py", line 37, in decode_event_data
    indexed_values = iter(decode_indexed_topics(tuple(n for n, i in inputs if i), topics))
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/dipdup/indexes/evm_subsquid_events/matcher.py", line 26, in decode_indexed_topics
    return decode_abi(indexed_inputs, indexed_bytes)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/codec.py", line 162, in decode
    return cast(Tuple[Any, ...], decoder(stream))
                                 ^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 131, in __call__
    return self.decode(stream)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_utils/functional.py", line 47, in inner
    return callback(fn(*args, **kwargs))
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 224, in decode
    self.validate_pointers(stream)
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 218, in validate_pointers
    decoder(stream)
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 131, in __call__
    return self.decode(stream)
           ^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 250, in decode
    raw_data = self.read_data_from_stream(stream)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/vscode/.local/lib/python3.11/site-packages/eth_abi/decoding.py", line 389, in read_data_from_stream
    raise InsufficientDataBytes(
eth_abi.exceptions.InsufficientDataBytes: Tried to read 32 bytes, only got 0 bytes.
________________________________________________________________________________

An unexpected error has occurred! Most likely it's a framework bug.
```
In this example:
- the indexer safely matches Mint event 374 here: https://etherscan.io/tx/0xb88f515a1b488757bb0f3c7b28280a60da942f4e9dbc495385b1862a6837f723#eventlog
- the indexer errors matching Mint event 27 here: https://etherscan.io/tx/0x9796d907d244aa853f0091cfa1435eeebfa47d244cbb8ec921861978c238bc97#eventlog

*Potential fix*

I inserted the following code in matcher.py to make an additional check on the number of topics:

```
inputs = package.get_converted_abi(typename)['events'][handler_config.name]['inputs']
abi_topic_count = sum(value for _, value in inputs if isinstance(value, bool))
if len(event.topics) != abi_topic_count + 1:
    continue
```

### Environment

* **Python version:** Python 3.11.4
* **DipDup version or commit hash:** dipdup, version 7.5.6
* **Database engine version:** n/a
* **Hasura version:** n/a
* **Are you in docker?** No
* **Can you reproduce this issue on `next`?** Haven't tried, but the matcher code is the same so suspect the bug exists there too

**Comment by droserasprout** [comment]
@scottincrypto, fixed in 7.5.7. Thanks for such a detailed report!

**Comment by scottnoidea** [comment]
My pleasure, thanks for the prompt fix!
